src/widget/wtrackwidgetgroup.cpp

```diff
@@ -0,0 +1,125 @@
+
+#include "widget/wtrackwidgetgroup.h"
+
+#include <QDebug>
+#include <QStylePainter>
+#include <QUrl>
+
+#include "control/controlobject.h"
+#include "util/dnd.h"
+#include "widget/wtrackmenu.h"
+
+namespace {
+
+int kDefaultTrackColorAlpha = 255;
+
+const WTrackMenu::Features kTrackMenuFeatures =
+        WTrackMenu::Feature::Playlist |
+        WTrackMenu::Feature::Crate |
+        WTrackMenu::Feature::Metadata |
+        WTrackMenu::Feature::Reset |
+        WTrackMenu::Feature::BPM |
+        WTrackMenu::Feature::Color |
+        WTrackMenu::Feature::FileBrowser |
+        WTrackMenu::Feature::Properties;
+
+} // anonymous namespace
+
+WTrackWidgetGroup::WTrackWidgetGroup(QWidget* pParent,
+        UserSettingsPointer pConfig,
+        TrackCollectionManager* pTrackCollectionManager,
+        const QString& group)
+        : WWidgetGroup(pParent),
+          m_group(group),
+          m_pConfig(pConfig),
+          m_trackColorAlpha(kDefaultTrackColorAlpha),
+          m_pTrackMenu(make_parented<WTrackMenu>(
+                  this, pConfig, pTrackCollectionManager, kTrackMenuFeatures)) {
+    setAcceptDrops(true);
+}
+
+WTrackWidgetGroup::~WTrackWidgetGroup() {
+}
+
+void WTrackWidgetGroup::setup(const QDomNode& node, const SkinContext& context) {
+    WWidgetGroup::setup(node, context);
+
+    bool ok = false;
+    int trackColorAlpha = context.selectInt(
+            node,
+            QStringLiteral("TrackColorAlpha"),
+            &ok);
+    if (ok && trackColorAlpha >= 0 && trackColorAlpha <= 255) {
+        m_trackColorAlpha = trackColorAlpha;
+    }
+}
+
+void WTrackWidgetGroup::slotTrackLoaded(TrackPointer pTrack) {
+    if (!pTrack) {
+        return;
+    }
+    m_pCurrentTrack = pTrack;
+    connect(pTrack.get(),
+            &Track::changed,
+            this,
+            &WTrackWidgetGroup::slotTrackChanged);
+    updateColor();
+}
+
+void WTrackWidgetGroup::slotLoadingTrack(TrackPointer pNewTrack, TrackPointer pOldTrack) {
+    Q_UNUSED(pNewTrack);
+    Q_UNUSED(pOldTrack);
+    if (m_pCurrentTrack) {
+        disconnect(m_pCurrentTrack.get(), nullptr, this, nullptr);
+    }
+    m_pCurrentTrack.reset();
+    updateColor();
+}
+
+void WTrackWidgetGroup::slotTrackChanged(TrackId trackId) {
+    Q_UNUSED(trackId);
+    updateColor();
+}
+
+void WTrackWidgetGroup::updateColor() {
+    if (m_pCurrentTrack) {
+        m_trackColor = mixxx::RgbColor::toQColor(m_pCurrentTrack->getColor());
+        m_trackColor.setAlpha(m_trackColorAlpha);
```

Comment:
Setting the alpha channel on an invalid color does not make any sense. Did you verify that this works as expected and m_trackColor is still invalid then? Please add a debug assertion to check this:

`DEBUG_ASSERT(m_trackColor.isValid() == m_pCurrentTrack->getColor().has_value());`

If has_value() is not supported on all platforms we may either remove it (implicit cast, not sure if this works) or replace it with a `static_cast<bool>`.